pykokkos/lib/ufuncs.py

```diff
@@ -1429,3 +1431,172 @@ def isinf(view):
                         view=view,
                         out=out)
     return out
+
+@pk.workunit
+def equal_impl_1d_double(tid: int,
+                         view1: pk.View1D[pk.double],
+                         view2: pk.View1D[pk.double],
+                         view2_size: int,
+                         view_result: pk.View1D[pk.uint16]):
+    view2_idx: int = 0
+    if view2_size == 1:
+        view2_idx = 0
+    else:
+        view2_idx = tid
+    if view1[tid] == view2[view2_idx]:
+        view_result[tid] = 1
+    else:
+        view_result[tid] = 0
+
+
+@pk.workunit
+def equal_impl_1d_uint16(tid: int,
+                         view1: pk.View1D[pk.uint16],
+                         view2: pk.View1D[pk.uint16],
+                         view2_size: int,
+                         view_result: pk.View1D[pk.uint16]):
+    view2_idx: int = 0
+    if view2_size == 1:
+        view2_idx = 0
+    else:
+        view2_idx = tid
+    if view1[tid] == view2[view2_idx]:
+        view_result[tid] = 1
+    else:
+        view_result[tid] = 0
+
+
+@pk.workunit
+def equal_impl_1d_int16(tid: int,
+                         view1: pk.View1D[pk.int16],
+                         view2: pk.View1D[pk.int16],
+                         view2_size: int,
+                         view_result: pk.View1D[pk.uint16]):
+    view2_idx: int = 0
+    if view2_size == 1:
+        view2_idx = 0
+    else:
+        view2_idx = tid
+    if view1[tid] == view2[view2_idx]:
+        view_result[tid] = 1
+    else:
+        view_result[tid] = 0
+
+
+@pk.workunit
+def equal_impl_1d_int32(tid: int,
+                         view1: pk.View1D[pk.int32],
+                         view2: pk.View1D[pk.int32],
+                         view2_size: int,
+                         view_result: pk.View1D[pk.uint16]):
+    view2_idx: int = 0
+    if view2_size == 1:
+        view2_idx = 0
+    else:
+        view2_idx = tid
+    if view1[tid] == view2[view2_idx]:
+        view_result[tid] = 1
+    else:
+        view_result[tid] = 0
+
+
+@pk.workunit
+def equal_impl_1d_int64(tid: int,
+                         view1: pk.View1D[pk.int64],
+                         view2: pk.View1D[pk.int64],
+                         view2_size: int,
+                         view_result: pk.View1D[pk.uint16]):
+    view2_idx: int = 0
+    if view2_size == 1:
+        view2_idx = 0
+    else:
+        view2_idx = tid
+    if view1[tid] == view2[view2_idx]:
+        view_result[tid] = 1
+    else:
+        view_result[tid] = 0
+
+def equal(view1, view2):
+    # TODO: write even more dispatching for cases where view1 and view2
+    # have different, but comparable, types (like float32 vs. float64?)
+    # this may "explode" without templating
```

Comment:
I think this is fine for now. I'm still undecided on how to properly fix this, but I'm leaning towards getting rid of the type annotations.